src/Nixfmt/Predoc.hs

```diff
@@ -600,9 +600,13 @@ layoutGreedy tw doc = Text.concat $ evalState (go [Group RegularG doc] []) (0, s
         mapStateT (Just . runIdentity) $
           go prio (unexpandSpacing post ++ rest)
       -- Try to render post onto one line
-      postRendered <- goGroup post rest
+      postRendered <- goGroup (fixListConcat post) rest
       -- If none of these failed, put together and return
       return (preRendered ++ prioRendered ++ postRendered)
+      where
+        -- Special case where list concatenation doesn't fit one line we need to replace space with newline
+        fixListConcat (Spacing Space : xs@(Text _ _ _ "++" : _)) = newline <> xs
```

Comment:
Please don't. This is a strong violation of layers and *really* should be fixed earlier in the data flow

Reply:
Yeah, it doesn't work anyway